app/code/core/Mage/Api/Model/Wsdl/Config.php

```diff
@@ -121,20 +123,22 @@ public function init()
              * Exclude Mage_Api wsdl xml file because it used for previous version
              * of API wsdl declaration
              */
-            $mergeWsdl->addLoadedFile('app/code/core/Mage/Api/etc/wsi.xml');
+            $mergeWsdl->addLoadedFile(mahoFindFileInIncludePath("$moduleDir/wsi.xml"));
+
+            // Base wsi file
+            $this->loadFile(mahoFindFileInIncludePath("$moduleDir/wsi.xml"));
```

Comment:
mmm :-\

Reply:
I know you changed that previously, but it introduced a bug.
Notice that the `app/code/core/Mage/Api/etc/wsdl.xml` file is included in the main branch.

```
# OM
/home/www-data/work/openmage/app/code/core/Mage/Api/etc/wsdl2.xml
/home/www-data/work/openmage/app/code/core/Mage/Core/etc/wsdl.xml
/home/www-data/work/openmage/app/code/core/Mage/Directory/etc/wsdl.xml
/home/www-data/work/openmage/app/code/core/Mage/Customer/etc/wsdl.xml
/home/www-data/work/openmage/app/code/core/Mage/Catalog/etc/wsdl.xml
/home/www-data/work/openmage/app/code/core/Mage/Sales/etc/wsdl.xml
/home/www-data/work/openmage/app/code/core/Mage/CatalogInventory/etc/wsdl.xml
/home/www-data/work/openmage/app/code/core/Mage/Checkout/etc/wsdl.xml
/home/www-data/work/openmage/app/code/core/Mage/Tag/etc/wsdl.xml
/home/www-data/work/openmage/app/code/core/Mage/GiftMessage/etc/wsdl.xml
/home/www-data/work/openmage/app/code/core/Mage/Downloadable/etc/wsdl.xml

# Maho Main
app/code/core/Mage/Api/etc/wsdl2.xml
/home/www-data/work/maho/git/app/code/core/Mage/Core/etc/wsdl.xml
/home/www-data/work/maho/git/app/code/core/Mage/Directory/etc/wsdl.xml
/home/www-data/work/maho/git/app/code/core/Mage/Customer/etc/wsdl.xml
/home/www-data/work/maho/git/app/code/core/Mage/Catalog/etc/wsdl.xml
/home/www-data/work/maho/git/app/code/core/Mage/Sales/etc/wsdl.xml
/home/www-data/work/maho/git/app/code/core/Mage/CatalogInventory/etc/wsdl.xml
/home/www-data/work/maho/git/app/code/core/Mage/Checkout/etc/wsdl.xml
/home/www-data/work/maho/git/app/code/core/Mage/Tag/etc/wsdl.xml
/home/www-data/work/maho/git/app/code/core/Mage/GiftMessage/etc/wsdl.xml
----> /home/www-data/work/maho/git/app/code/core/Mage/Api/etc/wsdl.xml
/home/www-data/work/maho/git/app/code/core/Mage/Downloadable/etc/wsdl.xml

# Maho PR-40
/home/www-data/work/maho/git/app/code/core/Mage/Api/etc/wsdl2.xml
/home/www-data/work/maho/git/app/code/core/Mage/Core/etc/wsdl.xml
/home/www-data/work/maho/git/app/code/core/Mage/Directory/etc/wsdl.xml
/home/www-data/work/maho/git/app/code/core/Mage/Customer/etc/wsdl.xml
/home/www-data/work/maho/git/app/code/core/Mage/Catalog/etc/wsdl.xml
/home/www-data/work/maho/git/app/code/core/Mage/Sales/etc/wsdl.xml
/home/www-data/work/maho/git/app/code/core/Mage/CatalogInventory/etc/wsdl.xml
/home/www-data/work/maho/git/app/code/core/Mage/Checkout/etc/wsdl.xml
/home/www-data/work/maho/git/app/code/core/Mage/Tag/etc/wsdl.xml
/home/www-data/work/maho/git/app/code/core/Mage/GiftMessage/etc/wsdl.xml
/home/www-data/work/maho/git/app/code/core/Mage/Downloadable/etc/wsdl.xml
```

Reply:
It is because in `app/code/core/Mage/Api/Model/Wsdl/Config/Base.php` we keep track of files already added, and by not using `mahoFindFileInIncludePath()` then then this line:
```php
$mergeWsdl->addLoadedFile('app/code/core/Mage/Api/etc/wsi.xml');
```

has no effect when we try to load the full path in `loadModulesConfiguration()`

Reply:
I didn't understand the problem of the last commit 😅 but I'll check it out with a bit more time, is it possible that there's code out there that calls the `loadFile`? because it would stop working right? I know we're breaking a lot of things anyway but I liked that the new loading magic would search files in a compatible way with previous code

Reply:
It shouldn't break anything. I thought pretty long about it. Because here:

https://github.com/MahoCommerce/maho/blob/9f57164453a1adb409d627875620a4e12f207848/lib/Varien/Simplexml/Config.php#L425-L435

If the 3rd party module didn't use `mahoFindFileInIncludePath()` (which they likely didn't) then the file won't be found, because we are trying to load from `/htdocs/public` and no XML files are there. So we will use `mahoFindFileInIncludePath()` then.

It just saves double path lookups because most of the time in the core code we are passing absolute paths, so we already ran `mahoFindFileInIncludePath()`